src/web.rs

```diff
@@ -138,17 +140,57 @@ impl WebImpl {
     }
 
     fn present_with_damage(&mut self, damage: &[Rect]) -> Result<(), SoftBufferError> {
-        let (width, _height) = self
+        let (buffer_width, _buffer_height) = self
             .size
             .expect("Must set size of surface before calling `present_with_damage()`");
+
+        let mut damage_iter = damage.iter();
+
+        let first_rect = damage_iter.next().expect("at least one damage rectangle");
```

Comment:
On second thought, maybe this should not panic, because it’s actually possible for someone to write this code:
```rust
surface.present_with_damage(&[])?;
```

Reply:
Sorry for the force-push here, just fixed the compile error.